atomic_reactor/util.py

```diff
@@ -626,3 +627,54 @@ def get_credentials(self, docker_registry):
         except KeyError:
             logger.warn('%s not found in .dockercfg', docker_registry)
             return {}
+
+
+def get_manifest_digests(image, insecure=False, dockercfg_path=None,
+                         versions=('v1', 'v2')):
+    """Return manifest digest for image.
+
+    :param image: ImageName, the remote image to inspect
+    :param insecure: bool, when True registry's cert is not verified
+    :param dockercfg_path: str, dirname of .dockercfg location
+    :param versions: tuple, which manifest schema versions to fetch digest
+
+    :return: dict, versions mapped to their digest
+    """
+    auth = None
+    if dockercfg_path:
+        dockercfg = Dockercfg(dockercfg_path).get_credentials(image.registry)
+
+        username = dockercfg.get('username')
+        password = dockercfg.get('password')
+        if username and password:
+            auth = requests.auth.HTTPBasicAuth(username, password)
+
+    context = '/'.join([x for x in [image.namespace, image.repo] if x])
+    tag = image.tag or 'latest'
+    url = 'https://{}/v2/{}/manifests/{}'.format(image.registry, context, tag)
```

Comment:
delete_from_registry is looking at the configuration it was given which is meant to include the URI scheme to use for the registry. The pulp_sync plugin also needs to know the URI scheme to use for the registry in order to tell it to Pulp.

Here, 'image.registry' is the registry name for use by Docker, and Docker generally guesses (tries https and then http).

So either we should do the same try-https-then-http thing, or we assume https and don't allow it to be overridden, or else we'll need to be told the scheme somehow.

If we assume https and don't allow configuration for it, we should probably note in the documentation that v2 registries must be accessible via https.